adapter/internal/operator/controllers/dp/api_controller.go

```diff
@@ -210,41 +227,48 @@ func (apiReconciler *APIReconciler) Reconcile(ctx context.Context, req ctrl.Requ
 		loggers.LoggerAPKOperator.Warnf("Api CR related to the reconcile request with key: %s returned error. Assuming API is already deleted, hence ignoring the error : %v", err)
 		return ctrl.Result{}, nil
 	}
-
-	var prodHTTPRoute1, sandHTTPRoute1 []string
-	if len(apiDef.Spec.Production) > 0 {
-		prodHTTPRoute1 = apiDef.Spec.Production[0].HTTPRouteRefs
-	}
-	if len(apiDef.Spec.Sandbox) > 0 {
-		sandHTTPRoute1 = apiDef.Spec.Sandbox[0].HTTPRouteRefs
-	}
-
-	// Retrieve HTTPRoutes
-	prodHTTPRoute, sandHTTPRoute, err := apiReconciler.resolveAPIRefs(ctx, prodHTTPRoute1,
-		sandHTTPRoute1, req.NamespacedName.String(), req.Namespace)
-	if err != nil {
-		loggers.LoggerAPKOperator.Warnf("Error retrieving ref CRs for API in namespace : %s, %v", err)
+	if apiState, err := apiReconciler.resolveAPIRefs(ctx, apiDef, req.NamespacedName, req.Namespace); err != nil {
+		loggers.LoggerAPKOperator.Warnf("Error retrieving ref CRs for API in namespace : %s, %v", req.NamespacedName.String(), err)
 		return ctrl.Result{}, err
+	} else if apiState != nil {
+		*apiReconciler.ch <- *apiState
 	}
-	loggers.LoggerAPKOperator.Debugf("HTTPRoutes are retrieved successfully for API CR %s", req.NamespacedName.String())
+	return ctrl.Result{}, nil
+}
 
-	if !apiDef.Status.Accepted {
-		apiState := apiReconciler.ods.AddAPIState(apiDef, prodHTTPRoute, sandHTTPRoute)
-		*apiReconciler.ch <- synchronizer.APIEvent{EventType: constants.Create, Event: apiState}
-		//TODO(amali) update status only after deployed without errors
-		apiReconciler.handleStatus(req.NamespacedName, constants.DeployedState, []string{})
-	} else if cachedAPI, events, updated :=
-		apiReconciler.ods.UpdateAPIState(&apiDef, prodHTTPRoute, sandHTTPRoute); updated {
-		*apiReconciler.ch <- synchronizer.APIEvent{EventType: constants.Update, Event: cachedAPI}
-		apiReconciler.handleStatus(req.NamespacedName, constants.UpdatedState, events)
+// applyStartupAPIs applies the APIs which are already available in the cluster at the startup of the operator.
+func (apiReconciler *APIReconciler) applyStartupAPIs() {
+	ctx := context.Background()
```

Comment:
we need to call `setCallApplyStartupAPIs()` as the first line to avoid other reconcile events calling this again